File: app/src/main/java/com/example/client/models/transaction/TransactionModel.java
package com.example.client.models.transaction;

public class TransactionModel {
    private int id;
    private int user;
    private int subject;
    private String ordercode;
    private Double amount;
    private String createAt;
    private int status;
    public TransactionModel(){

    }

    public TransactionModel(int id, int user, int subject, String ordercode, Double amount, String createAt, int status) {
        this.id = id;
        this.user = user;
        this.subject = subject;
        this.ordercode = ordercode;
        this.amount = amount;
        this.createAt = createAt;
        this.status = status;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getUser() {
        return user;
    }

    public void setUser(int user) {
        this.user = user;
    }

    public int getSubject() {
        return subject;
    }

    public void setSubject(int subject) {
        this.subject = subject;
    }

    public String getOrdercode() {
        return ordercode;
    }

    public void setOrdercode(String ordercode) {
        this.ordercode = ordercode;
    }

    public int getStatus() {
        return status;
    }

    public void setStatus(int status) {
        this.status = status;
    }

    public Double getAmount() {
        return amount;
    }

    public void setAmount(Double amount) {
        this.amount = amount;
    }

    public String getCreateAt() {
        return createAt;
    }

    public void setCreateAt(String createAt) {
        this.createAt = createAt;
    }


}

File: app/src/main/java/com/example/client/models/transaction/TransactionModel.kt
package com.example.client.models.transaction

import com.example.client.app.Constants
import com.example.client.base.BaseModel

data class TransactionModel(
        var transid: String,
        var transid_momo: String?,
        var created_at: String,
        var type: Constants.TransactionType,
        var amount: Double,
        var user_id: Int,
        var status: Int,
        var order_code: String?,
) : BaseModel()

data class TransactionResponse(
        var transid: String?,
        var transid_momo: String?,
        var created_at: String?,
        var type: Constants.TransactionType?,
        var amount: Double?,
        var user_id: Int?,
        var status: Int?,
        var order_code: String?,
) : BaseModel() {
    fun toTransactionModel() = TransactionModel(
            transid = transid.orEmpty(),
            transid_momo = transid_momo,
            created_at = created_at.orEmpty(),
            type = type ?: Constants.TransactionType.PAID,
            amount = amount ?: 0.0,
            user_id = user_id ?: -1,
            status = status ?: 0,
            order_code = order_code
    )
}

fun List<TransactionResponse>.toTransactions() = map { it.toTransactionModel() }

data class DataTransactionResponse(val order_code: String)
